src/utils.ts

```diff
@@ -173,3 +174,39 @@ export const deepEqual = (obj1: any, obj2: any): boolean => {
 
   return true;
 };
+
+export const getColumnNameByCase = (columnName: string) => {
+  if (isQuotedIdentifier(columnName)) {
+    return columnName;
+  }
+  const showColumnNamesInLowercase = workspace
+    .getConfiguration("dbt")
+    .get<boolean>("showColumnNamesInLowercase", true);
+  return showColumnNamesInLowercase ? columnName.toLowerCase() : columnName;
+};
+
+export const isColumnNameEqual = (
+  columnNameFromYml: string | undefined,
+  columnNameFromDb: string | undefined,
+) => {
+  if (!columnNameFromYml || !columnNameFromDb) {
+    return false;
+  }
+
+  if (columnNameFromYml === columnNameFromDb) {
+    return true;
+  }
+
+  const showColumnNamesInLowercase = workspace
+    .getConfiguration("dbt")
+    .get<boolean>("showColumnNamesInLowercase", true);
+
+  if (showColumnNamesInLowercase) {
+    return columnNameFromYml.toLowerCase() === columnNameFromDb.toLowerCase();
+  }
+
+  return false;
+};
+
+export const isQuotedIdentifier = (name: string) =>
+  /^\d|(?=.*[a-z])(?=.*[A-Z])|[- "']/.test(name);
```

Comment:
I think this is the wrong approach. You should test if is an unquoted identifier using the `UNQUOTED_CASE_INSENSITIVE = re.compile("^([_A-Za-z]+[_A-Za-z0-9$]*)$")`

Reply:
modified using this regex